File: frontend/src/types/widget.ts
export const WidgetType = {
  LINE_CHART: 'LINE_CHART',
  BAR_CHART: 'BAR_CHART',
  TEXT: 'TEXT'
} as const;

export type WidgetType = typeof WidgetType[keyof typeof WidgetType];

export interface Widget {
  id: string;
  type: WidgetType;
  position: number;
  data?: string;
  createdAt: string;
  updatedAt: string;
}

File: api/Models/Widget.cs
namespace Api.Models;

public class Widget
{
    public Guid Id { get; set; }
    public Guid DashboardId { get; set; }
    public WidgetType Type { get; set; }
    public int Position { get; set; }
    public string? Data { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public Dashboard? Dashboard { get; set; }
}

